### ISSUE TITLE
Is there any way to run MCC on snapshots or pre-releases?

### ISSUE THREAD

**Issue by EchoesNetwork** [ORelio/Minecraft-Console-Client#507]
Hi, I'm just curious if this is possible. All my servers run on the 1.13-pre3. Is there some way to add a new profile for the bot so that it'll run on the latest version of MC?

thanks.

**Comment by Pokechu22** [comment]
It would require work to update it to use the new protocol.  I believe I have a `1.13-support` branch on my fork that's updated to ... 17w50a, it seems.  No idea how stable it is.  In any case, there are several things that make it hard to update:

* I stopped focusing on the bug that required updating it each time so I could automate the tests (that's the most important reason for why I haven't been working on it)
* The new tab completion system requires parsing [declare commands](http://wiki.vg/Pre-release_protocol#Declare_Commands).  I tried to work around this by sending a different packet to the server, but I don't recall how well it works (and I think I ran into weird synchronization issues)
* Most importantly, [the flattening](https://minecraft.gamepedia.com/The_Flattening) happened, and that means that the assumptions made about block IDs and metadata no longer hold.  That one I haven't made any attempts to fix; the since-introduced data generators (still undocumented, but briefly mentioned on [pre-release protocol§§§Global palette](http://wiki.vg/Pre-release_protocol#Global_palette)) may make it slightly more feasible though.

For other snapshots, it wasn't super hard (if there were no protocol changes, you just need to add the protocol version into the list), but for these there are several challenges.
